src/NServiceBus.RavenDB/Internal/TransactionRecoveryStorageVerifier.cs

```diff
@@ -0,0 +1,25 @@
+﻿namespace NServiceBus.RavenDB.Internal
+{
+    using Raven.Client;
+    using Raven.Client.Document;
+    using Raven.Client.Document.DTC;
+
+    static class TransactionRecoveryStorageVerifier
+    {
+        public static void ReplaceStorageIfNotSetByUser(IDocumentStore store)
+        {
+            var docStore = store as DocumentStore;
+
+            if (docStore == null)
+                return;
+
+            var currentStorage = docStore.TransactionRecoveryStorage;
+
+            // VolatileOnlyTransactionRecoveryStorage is the default. If set to anything else, don't change it!
+            if (currentStorage == null || currentStorage is VolatileOnlyTransactionRecoveryStorage)
```

Comment:
And I think we shouldn't allow a user to make a dumb choice when dealing
with our persistence, or at least log a big warning stating that is super
unsafe.

On Thu, Jan 28, 2016, 18:58 David Boike <notifications@github.com> wrote:

> In src/NServiceBus.RavenDB/Internal/TransactionRecoveryStorageVerifier.cs
> <https://github.com/Particular/NServiceBus.RavenDB/pull/134#discussion_r51159441>
> :
>
> > +    using Raven.Client.Document;
> > +    using Raven.Client.Document.DTC;
> > +
> > +    static class TransactionRecoveryStorageVerifier
> > +    {
> > +        public static void ReplaceStorageIfNotSetByUser(IDocumentStore store)
> > +        {
> > +            var docStore = store as DocumentStore;
> > +
> > +            if (docStore == null)
> > +                return;
> > +
> > +            var currentStorage = docStore.TransactionRecoveryStorage;
> > +
> > +            // VolatileOnlyTransactionRecoveryStorage is the default. If set to anything else, don't change it!
> > +            if (currentStorage == null || currentStorage is VolatileOnlyTransactionRecoveryStorage)
>
> VolatileOnlyTransactionRecoveryStorage is actually the default for a
> DocumentStore. While it could (potentially, possibly?) be null, and that's
> included here for completeness against NRE, there's no way to tell if a
> user selected VolatileOnlyTransactionRecoveryStorage on purpose or is just
> accepting the default.
>
> But either way, it's a dumb choice. It's the DTC transaction recovery
> equivalent of in-memory persistence.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/NServiceBus.RavenDB/pull/134/files#r51159441>
> .
>
-- 
Mauro Servienti | Solution Architect | Particular Software